corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -207,11 +210,46 @@ def as_string(self):
         finally:
             self.payload.close()
 
+    def get_content_length(self):
+        if self.payload_path:
+            return os.path.getsize(self.payload_path)
+        return None
+
+    def as_file(self):
+        return self.payload
+
+    def finalize(self):
+        # When serving a cached initial payload we should still generate a new sync log
+        # This is to avoid issues with multiple devices ending up syncing to the same
+        # sync log, which causes all kinds of assertion errors when the two devices
+        # touch the same cases
+        if self and self.is_initial:
+            # read up to
+            try:
+                file_reference = copy_payload_and_synclog_and_get_new_file(self.payload)
+                self.payload = file_reference.file
+                self.payload_path = file_reference.path
+            except Exception, e:
+                # don't fail hard if anything goes wrong since this is an edge case optimization
+                soft_assert(to=['czue' + '@' + 'dimagi.com'])(False, u'Error finalizing cached log: {}'.format(e))
+
+
+class CachedResponse(object):
+    def __init__(self, payload):
+        self.payload = payload
+
+    def exists(self):
```

Comment:
other things were calling `.exists` on this and i didn't want to chase down all the references (I didn't change the api of this class at all)